enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,441 @@
+The only difference for heterogeneous payloads though is that, the extract command would return a set of tools for all supported payload architectures
+when the `--command-os=*` flag is used. The table below describes the combination of os/architectures for which the oc client and the installer binary
+would be extracted.
+
+| Client platform |	Image arch | Image | Path | Tarball |
+| :--- | :--- | :--- | :--- | :--- |
+| linux/amd64 |	amd64 |	cli | /usr/bin/oc |	openshift-client-linux-amd64-%s.tar.gz |
+| linux/arm64	| arm64	| cli	| /usr/bin/oc | openshift-client-linux-arm64-%s.tar.gz |
+| linux/ppc64le | ppc64le | cli | /usr/bin/oc | openshift-client-linux-ppc64le-%s.tar.gz |
+| linux/s390x | s390x | cli | /usr/bin/oc | openshift-client-linux-s390x-%s.tar.gz |
+| darwin/amd64 | amd64* | cli-artifacts | /usr/share/openshift/mac/oc | openshift-client-mac-%s.tar.gz |
+| darwin/arm64 | arm64* |	cli-artifacts |	/usr/share/openshift/mac_arm64/oc | openshift-client-mac-arm64-%s.tar.gz |
+| windows/amd64 |	amd64* | cli-artifacts | /usr/share/openshift/windows/oc.exe | openshift-client-windows-%s.tar.gz |
+
+| Install platform | Image arch | Image |	Path | Tarball |
+| :--- | :--- | :--- | :--- | :--- |
+| linux/amd64 |	amd64 | installer |	/usr/bin/openshift-install | openshift-install-linux-amd64-%s.tar.gz |
+| linux/arm64	| arm64	| installer | /usr/bin/openshift-install | openshift-install-linux-arm64-%s.tar.gz |
+| linux/ppc64le |	ppc64le | installer | /usr/bin/openshift-install | openshift-install-linux-ppc64le-%s.tar.gz |
+| linux/s390x	| s390x	| installer	| /usr/bin/openshift-install | openshift-install-linux-s390x-%s.tar.gz |
+| darwin/amd64 | amd64* | installer-artifacts |	/usr/share/openshift/mac/openshift-install | openshift-install-mac-%s.tar.gz |
+| darwin/arm64 | arm64* | installer-artifacts | /usr/share/openshift/mac_arm64/openshift-install | openshift-install-mac-arm64-%s.tar.gz |
+
+*Technically it shouldn't matter which manifest's installer-artifacts are used, the windows and mac binaries are all cross-compiled anyway.
+
+### Upgrades - Cincinnati and CVO
+
+In addition to the homogeneous payload graph that we have today, [support][OCPPLAN-7642] needs to be added for a graph for heterogeneous
+payloads which CVO will need to look at to make upgrade recommendations. There are two upgrade scenarios to consider with the addition of
+heterogeneous payloads:
+
+- heterogeneous -> heterogeneous release payload upgrade after a fresh install
+- homogeneous -> heterogeneous release payload upgrade
+
+A separate heterogeneous graph will be introduced for each version starting from the version which supports heterogeneous payloads. A cluster
+installed as a day1 heterogeneous cluster with a heterogeneous payload will use the heterogeneous graph for further updates.
+
+An existing homogeneous cluster can be converted to a heterogeneous cluster through a migration process which is a special upgrade process. The
```

Comment:
why do we want the customers to configure it explicitly? Isnt it just a natural evolution that should be transparent to them?

Reply:
we want the migration to a heterogeneous payload be a separate lifecycle event so we can troubleshoot issues related to the transition better. Also, homogeneous payloads will continue existing alongside heterogeneous payloads, so until heterogeneous payloads become the de facto, we will have the migration be an explicit event.